src/Resources/Resources/ActiveDirectory/Cmdlets/NewAzureADServicePrincipalCommand.cs

```diff
@@ -327,11 +315,31 @@ private void CreateSimpleServicePrincipal()
                 }
             };
 
-            var shouldProcessMessage = SkipRoleAssignment() ?
-                                        string.Format("Adding a new service principal to be associated with an application " +
-                                                      "having AppId '{0}' with no permissions.", createParameters.ApplicationId) :
-                                        string.Format("Adding a new service principal to be associated with an application " +
+            var shouldProcessMessage = string.Format("Adding a new service principal to be associated with an application " +
+                                                      "having AppId '{0}' with no permissions.", createParameters.ApplicationId);
+
+            if (!SkipRoleAssignment())
+            {
+                var subscriptionId = DefaultContext.Subscription.Id;
```

Comment:
duplicated statement 1/2

Reply:
A `var subscriptionId = DefaultContext?.Subscription?.Id;` before the `if` block maybe?

Reply:
it's possible that DefaultContext doesn't have subscription, which will throw null reference if this line was not executed inside (!SkipRoleAssignment)